src/jitlayers.cpp

```diff
@@ -1629,6 +1629,10 @@ JuliaOJIT::JuliaOJIT()
     DLSymOpt(std::make_unique<DLSymOptimizer>(false)),
     ContextPool([](){
         auto ctx = std::make_unique<LLVMContext>();
+#ifndef JL_LLVM_OPAQUE_POINTERS
+        if (ctx->supportsTypedPointers())
+            ctx->setOpaquePointers(false);
+#endif
```

Comment:
```suggestion
        else
#endif
            ctx->setOpaquePointers(true);
```

Reply:
Isn't this redundant? `JL_LLVM_OPAQUE_POINTERS` is only true for LLVM 16+, and opaque pointers are the default starting from LLVM 15 on.

Reply:
Also, I think we're misunderstanding `supportsTypedPointers`, which returns whether typed pointers are being used in the context right now. So we should always call that function in `JL_LLVM_OPAQUE_POINTERS`.